extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -373,40 +369,45 @@ export class AmpStoryPage extends AMP.BaseElement {
   }
 
   /**
-   * Reads the storyNextUp param and sets it as the auto-advance-after attribute
-   * if this is the last page and there isn't a value set by the publisher.
+   * Reads the storyNextUp param if provided and sets the auto-advance-after
+   * attribute if there isn't a value set by the publisher. The
+   * auto-advance-after attribute will be set to the id of the first
+   * video if there is one, or the value of the storyNextUp param otherwise.
    * @private
    */
   maybeSetStoryNextUp_() {
     const autoAdvanceAttr = this.element.getAttribute('auto-advance-after');
+    // This is a private param used for testing, it may be changed
+    // or removed without notice.
     const storyNextUpParam = Services.viewerForDoc(this.element).getParam(
       'storyNextUp'
     );
-    if (
-      autoAdvanceAttr === null &&
-      storyNextUpParam !== null &&
-      this.isLastPage_()
-    ) {
+    if (autoAdvanceAttr !== null || storyNextUpParam === null) {
+      return;
+    }
+    const video = this.getFirstVideo_();
+    if (video !== null) {
       addAttributesToElement(
         this.element,
-        dict({'auto-advance-after': storyNextUpParam})
+        dict({'auto-advance-after': video.id})
```

Comment:
Videos are not required to have an ID, so this might give you `null` which would fail to create an auto-advance mode based on the video.

Let's write a small fallback for when there's no ID. It should be unique, so you can use the page ID (guaranteed to be unique, `this.element.id`) and add a prefix/suffix.

Reply:
Good call, this appears to be fairly common. I was able to verify that the feature works correctly locally with a story that has a mix of cases. 

Running `amp` then loading http://localhost:8000/examples/amp-story/ampconf.html#storyNextUp=6s ([ampconf.html](https://github.com/ampproject/amphtml/blob/583a7334e08971f55dc8a2710fe6ea2a60da0757/examples/amp-story/ampconf.html)) demonstrates this.